python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -303,47 +270,23 @@ def runAssociation(self, sensorRef):
 
             - apdb : `lsst.dax.apdb.Apdb` Initialized association database containing final association
                 results.
-            - taskResults : output of `config.associator.run` (`lsst.pipe.base.Struct`).
+            - taskResults : output of `config.diaPipe.run` (`lsst.pipe.base.Struct`).
         """
-        self.log.info("Running Association...")
-
         diffType = self.config.differencer.coaddName
 
-        catalog = sensorRef.get(diffType + "Diff_diaSrc")
-        diffim = sensorRef.get(diffType + "Diff_differenceExp")
-
-        diaSources = self.diaSourceDpddifier.run(catalog,
-                                                 diffim,
-                                                 return_pandas=True)
-        # Load the DiaObjects and DiaSource history. If this fails it is due
-        # to an improperly set up Apdb hence we don't want it caught by the
-        # try clause.
-        loaderResult = self.diaCatalogLoader.run(diffim, self.apdb)
-
         try:
-            results = self.associator.run(diaSources,
-                                          loaderResult.diaObjects,
-                                          loaderResult.diaSources)
-            diaForcedSources = self.diaForcedSource.run(
-                results.diaObjects,
-                sensorRef.get("ccdExposureId_bits"),
-                sensorRef.get("calexp"),
-                diffim)
-
-            # Store DiaSources, updated DiaObjects, and DiaForcedSources in the
-            # Apdb.
-            self.apdb.storeDiaSources(results.diaSources)
-            self.apdb.storeDiaObjects(
-                results.updatedDiaObjects,
-                diffim.getInfo().getVisitInfo().getDate().toPython())
-            self.apdb.storeDiaForcedSources(diaForcedSources)
+            results = self.diaPipe.run(
+                diaSourceCat=sensorRef.get(diffType + "Diff_diaSrc"),
+                diffIm=sensorRef.get(diffType + "Diff_differenceExp"),
+                exposure=sensorRef.get("calexp"),
+                ccdExposureIdBits=sensorRef.get("ccdExposureId_bits"))
         finally:
             # apdb_marker triggers metrics processing; let them try to read
             # something even if association failed
-            sensorRef.put(self.config.apdb.value, "apdb_marker")
+            sensorRef.put(results.apdb_marker, "apdb_marker")
```

Comment:
Happy to remove the `try` block. Wasn't completely sure why it was there so I kept it around. Is the `try` block something we should have or is completely okay to remove?

Reply:
Removing it means that `ap_verify` will no longer try to get metrics out of the database if `diaPipe` fails partway through, which might make debugging harder. That would be the only thing we lose.

Reply:
Would it work to put the `try` clause into the appropriate place of `diaPipe`?

Reply:
In theory it could go into `runQuantum`, but depending on how the middleware group decides to deal with tasks throwing exceptions (e.g., "pruning" downstream task runs), it might not make any difference. Not sure it's worth doing now.

Reply:
Okay, I'll leave it for now and merge once the new Jenkins run passes.